src/main.rs

```diff
@@ -1,34 +1,28 @@
 #![feature(proc_macro_hygiene, decl_macro)]
 
-use std::env;
-use rocket::{self, routes, Config};
+mod config;
+mod pudeuko;
+mod api;
 
-mod domain;
-mod items_controller;
-mod pudeuko_service;
-mod dropbox_client;
+use rocket::{self, Rocket, routes};
+use pudeuko::pudeuko_service::PudeukoService;
 
-fn create_configuration() -> Config {
-    let mut config = Config::active().expect("Could not load configuration");
-    if let Ok(port_str) = env::var("PORT") {
-        let port = port_str.parse().expect("Could not parse PORT");
-        config.set_port(port);
-    }
-    config
-}
-
-fn main() {
-    let dropbox_token = env::var("DROPBOX_TOKEN").expect("You have to provide a Dropbox API token");
-    let dropbox_client = dropbox_client::DropboxClient::new(dropbox_token);
-    let pudeuko_service = pudeuko_service::PudeukoService::new(dropbox_client);
+fn rocket(port: u16, pudeuko_service: PudeukoService) -> Rocket {
+    let mut config = rocket::Config::active().expect("Could not load configuration");
+    config.set_port(port);
 
-    let config = create_configuration();
     rocket::custom(config)
         .manage(pudeuko_service)
         .mount("/items", routes![
-            items_controller::get_items,
-            items_controller::post_item,
-            items_controller::get_item,
+            api::handlers::get_items,
+            api::handlers::post_item,
+            api::handlers::get_item,
         ])
-        .launch();
+}
+
+fn main() {
+    let config = config::Config::load_from_env();
+    let pudeuko_service = PudeukoService::new(&config.dropbox_token);
```

Comment:
uuu, a dlaczego zrezygnowałeś z wstrzykiwania zależności? to chyba nie był taki zły pomysł?

Reply:
Cały czas jest w niezmienionej formie!

Reply:
Eee to mi chyba musisz wytłumaczyć przy okazji :D